### ISSUE TITLE
broken for archives

### ISSUE THREAD

**Issue by andreineculau** [wofr06/lesspipe.sh#57]
running lesspipe on a zip/tar.gz/etc prints nothing, leaving less to ask if I still want to see the file as binary

I've bisected the bad commit in

```
commit e0af27d9b382ce073f011366a5b5b0c3ed23f7cb (HEAD)
Author: Konfekt <Konfekt@users.noreply.github.com>
Date:   2021-05-11 17:20:49 +0200

    correctly detect image file

    addresses https://github.com/wofr06/lesspipe/issues/54
```

I'll investigate further.

**Comment by andreineculau** [comment]
Hard (for me) to asses the right fix, especially since the commit diff shows a many whitespace diffs as well.

What I can say is the culprit is the removal of these lines https://github.com/wofr06/lesspipe/commit/e0af27d9b382ce073f011366a5b5b0c3ed23f7cb?diff=unified#diff-1d1c7e00b6f9694b32709c3c2f27ce31a59fb99df5ebfd476c5b93206bd3c329L177-L179

but putting them back, means that #54 is back as well.

My 2 penny is that the fix should be reverted, a new version published, and then commit a correct fix for handling PNG.

**Comment by andreineculau** [comment]
this regression is not macos specific. tested on ubuntu 18.04

**Comment by wofr06** [comment]
Just fixed. 3 lines of code seemed to have disappeared unnoticed. The testsuite (make test) was failing all over the place. Now only two more exotic tests do fail.

**Comment by andreineculau** [comment]
Confirmed that I can list tarball contents again. Can you tag 1.87 please?

EDIT: Thanks!

**Comment by tiagoamartins** [comment]
Same issue happening with version 1.89. I have verified it in Arch Linux and macOS. It was working fine in version 1.88.

**Comment by andreineculau** [comment]
@wofr06 would you be interested in a PR to add a Github Actions workflow? I'm assuming at least that tests are failing one again because of the same 3 lines of code that "disappeared", though I see now that there's a comment/reason for that

https://github.com/wofr06/lesspipe/blob/b429098/lesspipe.sh#L179-L182

https://github.com/wofr06/lesspipe/issues/55#issuecomment-905410262

**Comment by m040601** [comment]
>  Basic things like less'ing gz files do not work and show up as binary. 


I can confirm this also happens, in Archlinux, v1.89.

The same files open perfectly using

gzip /usr/bin/zcat
gzip /usr/bin/zgrep
gzip /usr/bin/zless

**Comment by wofr06** [comment]
This was hopefully fixed already in 1.91, but the latest git shows with my test suite, that the bug has gone.
